src/FileSynchronizer.ts

```diff
@@ -0,0 +1,105 @@
+import EventEmitter from 'events';
+import { opendir, stat } from 'fs/promises';
+import { join, resolve, extname } from 'path';
+
+import { Minimatch } from 'minimatch';
+
+import {
+  SyncOptions,
+  WalkOptions,
+  WalkCallback,
+  FileInfo,
+  ProcessedPattern,
+} from './types';
+
+export declare interface FileSynchronizer {
+  emit(event: 'end'): boolean;
```

Comment:
do you see a use case for this event? I'm skeptical because there's no test for it :D